cmd/config.go

```diff
@@ -101,62 +137,162 @@ func createTemplates() error {
+
+	var interfaceNames []string
+
+	for _, inter := range interfaces {
+
+		if strings.Contains(inter.Name, "lo") || strings.Contains(inter.Name, "tun") || strings.Contains(inter.Name, "docker") {
+			continue
 		}
-		_, err = f.Write([]byte(bb))
+
+		interfaceNames = append(interfaceNames, inter.Name)
+	}
+
+	if l := len(interfaceNames); l > 1 {
+		prompt := promptui.Select{
+			Label: "Select Network Adapter",
+			Items: interfaceNames,
+		}
+
+		_, result, err := prompt.Run()
 		if err != nil {
 			return err
 		}
+
+		viper.Set("network", result)
+	} else if l == 1 {
+		viper.Set("network", interfaceNames[0])
+	} else {
+		//in the future we should probably create a host network like docker does.
+		return errors.New("No network interfaces available")
 	}
-	return nil
 
+	return nil
 }
 
-func createRelationsStore() error {
-	// create config storage directory and file
+func determineDefaultTemplate() error {
 
-	err := os.MkdirAll(filepath.Join(path.GetConfigPath(), "templates"), 0755)
+	prompt := promptui.Select{
+		Label: "Select default profile",
+		Items: profiles,
+	}
+
+	_, result, err := prompt.Run()
 	if err != nil {
 		return err
 	}
 
-	f, err := os.Create(filepath.Join(path.GetConfigPath(), "templates", "relations.yaml"))
+	viper.Set("template", result)
+	return nil
+}
+
+func determineUID() error {
+	curUser, err := user.Current()
 	if err != nil {
+		log.Error("Unable to get user from OS.")
 		return err
 	}
 
-	defer f.Close()
-
+	viper.Set("uid", curUser.Uid)
 	return nil
 }
-func init() {
-	rootCmd.AddCommand(configCmd)
 
-	// Here you will define your flags and configuration settings.
+func validateLxdSetup() error {
+	curUser, err := user.Current()
+	if err != nil {
+		log.Error("Unable to get user from OS.")
+		return err
+	}
+
+	lxdGroup, err := user.LookupGroup("lxd")
+	if err != nil {
+		log.Error("Unable to get lxd group Id from OS, are you sure lxd is installed?")
+		return err
+	}
 
-	// Cobra supports Persistent Flags which will work for this command
-	// and all subcommands, e.g.:
-	// configCmd.PersistentFlags().String("foo", "", "A help for foo")
+	userGroups, err := curUser.GroupIds()
+	if err != nil {
+		log.Error("Unable to get user's group IDs from OS.")
+		return err
+	}
 
-	// Cobra supports local flags which will only run when this command
-	// is called directly, e.g.:
-	configCmd.Flags().BoolP("create", "c", false, "Create global config file in $HOME")
-	configCmd.Flags().BoolP("templates", "t", false, "Create global template folders in $HOME")
-}
+	userInGroup := false
+
+	for _, gid := range userGroups {
+		if gid == lxdGroup.Gid {
+			userInGroup = true
+		}
+	}
 
-const configTemplate = `cliimage: "18.10"
-guiimage: "18.10"
-utilimage: "18.10"
-ethernet: enp5s0`
+	if !userInGroup {
+		log.Error(fmt.Sprintf("The current user %s is not in the 'lxd' group. Please add the user by running 'adduser %s lxd' in terminal then logging out and back in before rerunning init.", curUser.Name, curUser.Username))
```

Comment:
Thanks for this! My first impression of  `devlx` was having it fail because my user was not in the `lxd` group. This patch would make for a smoother experience.